packages/web5/src/dwn-api.ts

```diff
@@ -0,0 +1,169 @@
+import type { Web5Agent } from '@tbd54566975/web5-agent';
+import type {
+  MessageReply,
+  RecordsReadOptions,
+  RecordsQueryOptions,
+  RecordsWriteMessage,
+  RecordsWriteOptions,
+  RecordsDeleteOptions,
+  ProtocolsQueryOptions,
+  ProtocolsConfigureOptions,
+} from '@tbd54566975/dwn-sdk-js';
+
+import { DwnInterfaceName, DwnMethodName, DataStream } from '@tbd54566975/dwn-sdk-js';
+
+import { Record } from './record.js';
+import { dataToBytes } from './utils.js';
+
+export type RecordsWriteRequest = {
```

Comment:
So actually... more thoughts... "SDK" to me implies something heavier than a library. 

In the context of DWN, a library could be a specific implementation of something from the [spec](https://identity.foundation/decentralized-web-node/spec/). For example, say we wanted to use SQL for data storage, but we wanted the DX to follow the DWN interfaces & standards, then that's a library. 

An SDK, to me, usually offers the comprehensive feature set in order to build a simple "Hello World" app but also administrative, or "meta-layer" features which aren't directly applicable to the end-user experience. For example, say the developer wanted to implement a DWeb App which requires orchestration of many-nodes. Such a use case may not directly be defined by the spec, but is totally necessary to really leverage the value of the underlying standard.

I think about it in terms of AWS (or any of the cloud providers). There is the developer experience of _provisioning_ (or, as I'm putting it, "administrating") cloud resources, and then there is the developer experience of actually _using_ those cloud resources. Both together is an "SDK" but solely the latter is a "Library."

I still hold that building a singular artifact which defines solely the interfaces & types is worth consideration.

Idk there's also the matter of debate around a "reference implementation" and how the only realistic path forward is for everyone to use the reference implementation and not implement their own based on the spec. For example, in the context of Bitcoin, there are many implementations, but to this date, Bitcoin Core still holds something like 99% of all traffic. Because the spec is so sophisticated that building your own invariably leads to production issues caused by trying to reconcile differences. 

Idk, I suppose it's semantic-heavy... but perhaps valuable to us all the same.

These things are emergent. If `dwn-sdk-js` has grown heavy and difficult to move then it makes sense to do things like what we've done here.

Reply:
Anyways... I don't think anything above changes the course with what you're doing here w/ `web5-js`. I think there is some interesting context in what I've written around the broader DX strategy, and nomenclature specifically.